### PR TITLE
Added Jade template engine support

### PR DESCRIPTION
**Pull request by Kurtz1993**
### Here's an example how to use a jade template with emailjs
```javascript
  var email = require('emailjs');
  var server = email.server.connect({
    user: "username",
    password: "password",
    host: "smtp.your-email.com",
    port: 25, // Or your custom SMTP Port
    ssl:   true
  });
server.send({
    text: "Example email",
    from: "Example <test@test.com>",
    to: "example@example.com",
    subject: "Sending a jade template through email ;D",
    template: "./templates/hello.jade"
  }, function (err, message){
    console.log(err || message);
  });
```

### CODE REVIEW

File: smtp/client.js [hunk 1]
```diff
@@ -85,7 +85,7 @@ Client.prototype =
       if(!(msg instanceof message.Message) 
           && msg.from 
           && (msg.to || msg.cc || msg.bcc)
-          && (msg.text || this._containsInlinedHtml(msg.attachment)))
+          && (msg.text || this._containsInlinedHtml(msg.attachment) || msg.template))
```
Comment (narcisoguillen) [inline smtp/client.js]: I think you have an extra parenthesis at the end.

      (msg.text || this._containsInlinedHtml(msg.attachment) || msg.template)

Comment (Kurtz1993) [inline smtp/client.js]: It's not an extra parenthesis, it's the end of the if condition.
____________________
Luis Rogelio Hernández López
luis.hlopez@outlook.com
@Kurtz1993
Sent using CloudMagic
On Mon, Mar 02, 2015 at 12:06 pm, Narciso <notifications@github.com> wrote:In smtp/client.js:
> @@ -85,7 +85,7 @@ Client.prototype =
>        if(!(msg instanceof message.Message) 
>            && msg.from 
>            && (msg.to || msg.cc || msg.bcc)
> -          && (msg.text || this._containsInlinedHtml(msg.attachment)))
> +          && (msg.text || this._containsInlinedHtml(msg.attachment) || msg.template))

I think you have an extra parenthesis at the end.

  (msg.text || this._containsInlinedHtml(msg.attachment) || msg.template)

**Comment by jacoborus** [comment]
I'm using emailjs and not interested in built in jade support. I recommend you create another module that uses emailjs as a dependency.

Something like https://github.com/jacoborus/curlymail , that one uses mustaches instead jade

**Comment by eleith** [comment]
i like the idea. i also agree with @jacoborus that this would better support modularity if this was an independent module that depends on emailjs.

since emailjs can take in HTML, jade doesn't need to be a part of emailjs, but rather on the outside, processing the jade -> html and passing that string into the library.

by accepting it, it only allows for one flavor of templates and adds complexity to the library.

thanks for the idea though!

**Comment by Kurtz1993** [comment]
No prob, I'll be working on an independent module that depends on emailjs and supports multiple template engines as suggested.
Thanks for the comments!

**Pull request closed without merging**